### PR TITLE
Add set of scripts to support Jenkins internal CI

### PR DESCRIPTION
**Pull request by ddokupil**

### CODE REVIEW

**Comment by postrational** [comment]
Will we add a `Jenkinsfile` to this as well?

File: .ci/jenkins/runCI.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ******************************************************************************
+
+SCRIPT=$(readlink -f "$0")
+BASEDIR=$(dirname "$SCRIPT")
+
+cd $BASEDIR && cd ../..
+
+if docker images | grep base_image; then
```
Comment (postrational) [inline .ci/jenkins/runCI.sh]: The names `base_image` and `test_image` seem very generic, they may conflict with others.

File: .ci/jenkins/base_image.dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
+FROM ubuntu:16.04
+
+ARG http_proxy
+ARG https_proxy
+ENV http_proxy=${http_proxy} https_proxy=${https_proxy}
```
Comment (ldurka) [inline .ci/jenkins/base_image.dockerfile]: to be sure I would export also uppercase
https://unix.stackexchange.com/questions/212894/whats-the-right-format-for-the-http-proxy-environment-variable-caps-or-no-ca

File: .ci/jenkins/base_image.dockerfile [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+# nGraph dependencies
+RUN apt-get -y update && apt-get -y install git build-essential cmake clang-3.9 git curl zlib1g zlib1g-dev libtinfo-dev \
```
Comment (ldurka) [inline .ci/jenkins/base_image.dockerfile]: Add at the beginning 
```
apt update --fix-missing &&
```
and at the end  apt autoremove -y && apt clean all

File: .ci/jenkins/base_image.dockerfile [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+                                            python python3 python-pip python3-pip python-dev python3-dev python-virtualenv \
+                                            protobuf-compiler libprotobuf-dev
+
+RUN pip install --upgrade pip setuptools wheel && pip3 install --upgrade pip setuptools wheel
+
+WORKDIR /root
```
Comment (ldurka) [inline .ci/jenkins/base_image.dockerfile]: you can remove this workdir section and use 
```
git clone git@...repo.. /full/path/to/repo
```

File: .ci/jenkins/base_image.dockerfile [hunk 5]
```diff
@@ -0,0 +1,15 @@
+FROM ubuntu:16.04
```
Comment (ldurka) [inline .ci/jenkins/base_image.dockerfile]: add header

File: .ci/jenkins/runCI.sh [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+   echo Base image found so proceeding to build ngraph and onnx
+else
+   echo No base image found so need to create one
+   docker build --build-arg http_proxy=$http_proxy --build-arg https_proxy=$https_proxy -t base_image -f .ci/jenkins/base_image.dockerfile .
+fi
+
+docker build --build-arg BASE_IMAGE=base_image -t test_image -f .ci/jenkins/test_image.dockerfile .
+
+docker run test_image
```
Comment (ldurka) [inline .ci/jenkins/runCI.sh]: Consider explicitly add a container name. and after test remove the container.

File: .ci/jenkins/runCI.sh [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ******************************************************************************
+
+SCRIPT=$(readlink -f "$0")
+BASEDIR=$(dirname "$SCRIPT")
+
+cd $BASEDIR && cd ../..
```
Comment (ldurka) [inline .ci/jenkins/runCI.sh]: please add comment why you change directory

File: .ci/jenkins/runCI.sh [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+if docker images | grep base_image; then
+   echo Base image found so proceeding to build ngraph and onnx
+else
+   echo No base image found so need to create one
+   docker build --build-arg http_proxy=$http_proxy --build-arg https_proxy=$https_proxy -t base_image -f .ci/jenkins/base_image.dockerfile .
```
Comment (ldurka) [inline .ci/jenkins/runCI.sh]: Consider use docker-compose to simplify the interface

File: .ci/jenkins/base_image.dockerfile (hunk 2 above)
Comment (ddokupil) [reply .ci/jenkins/base_image.dockerfile]: done

File: .ci/jenkins/base_image.dockerfile (hunk 3 above)
Comment (ddokupil) [reply .ci/jenkins/base_image.dockerfile]: done

File: .ci/jenkins/base_image.dockerfile (hunk 4 above)
Comment (ddokupil) [reply .ci/jenkins/base_image.dockerfile]: done

File: .ci/jenkins/base_image.dockerfile (hunk 5 above)
Comment (ddokupil) [reply .ci/jenkins/base_image.dockerfile]: done

File: .ci/jenkins/runCI.sh (hunk 6 above)
Comment (ddokupil) [reply .ci/jenkins/runCI.sh]: done

File: .ci/jenkins/runCI.sh (hunk 7 above)
Comment (ddokupil) [reply .ci/jenkins/runCI.sh]: done

File: .ci/jenkins/runCI.sh (hunk 8 above)
Comment (ddokupil) [reply .ci/jenkins/runCI.sh]: Will use in next iterations.

File: .ci/jenkins/runCI.sh (hunk 1 above)
Comment (ddokupil) [reply .ci/jenkins/runCI.sh]: changed

**Comment by postrational** [comment]
Looks good. Please add a `Jenskinsfile` and enable testing on Jenkins.

**Pull request closed without merging**